packages/ses/test/test-import.js

```diff
@@ -480,6 +483,54 @@ test('module alias', async t => {
   // );
 });
 
+test('import reflexive module alias', async t => {
+  t.plan(1);
+
+  const makeImportHook = makeNodeImporter({
+    'https://example.com/index.js': `
+      import self from 'self';
+      export default 10;
+      t.is(self, 10);
+    `,
+  });
+
+  const wrappedImportHook = makeImportHook('https://example.com');
+
+  const importHook = async specifier => {
+    const candidates = [specifier, `${specifier}.js`, `${specifier}/index.js`];
+    for (const candidate of candidates) {
+      // eslint-disable-next-line no-await-in-loop
+      const record = await wrappedImportHook(candidate).catch(_ => undefined);
+      if (record !== undefined) {
+        return { record, specifier };
+      }
+    }
+    throw new Error(`Cannot find module ${specifier}`);
+  };
+
+  const moduleMapHook = specifier => {
+    if (specifier === 'self') {
+      // eslint-disable-next-line no-use-before-define
+      return compartment.module('./index.js');
+    }
+    return undefined;
+  };
+
+  const compartment = new Compartment(
+    {
+      t,
+    },
+    {},
+    {
```

Comment:
https://www.youtube.com/watch?v=FR4iRawfvNA

Reply:
So…I paid money to see this in a “theatre” and a friend of mine’s reaction was much better than mine: “I’m not sure whether I want my money back or to go back in and see it again.”

And, I guess this is indeed “nuts”.